backends/s3/s3.go

```diff
@@ -189,9 +182,38 @@ func (b *Backend) Store(ctx context.Context, name string, data []byte) error {
 		return err
 	}
 	if b.opt.UseUpdateMarker {
+		var wg sync.WaitGroup
+		wg.Add(1)
+
+		// Update cache
+		go func() {
+			defer wg.Done()
+			// Update size or add to local list if not present
+			l := b.lastList.Len()
+			idx := sort.Search(l, func(i int) bool { return b.lastList[i].Name >= name })
+			blob := simpleblob.Blob{Name: name, Size: int64(len(data))}
+			b.mu.Lock()
```

Comment:
Since this is a separate function and the distance between the Lock and Unlock is quite large, I would actually use `defer b.mu.Unlock()` here.

One big advantage of `defer` is that you can be sure it will still be run in case of an early return or panic. There is no such place now, but it's not immediately obvious without inspecting the code between the Lock and Unlock, and someone might inadvertently add one later.